crates/pop-cli/src/common/wallet.rs

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: GPL-3.0
+
+use crate::{
+	cli::traits::Cli,
+	wallet_integration::{FrontendFromString, TransactionData, WalletIntegrationManager},
+};
+use cliclack::{log, spinner};
+
+/// The prompt to ask the user if they want to use the wallet for signing.
+pub const USE_WALLET_PROMPT: &str = "Do you want to use your browser wallet to sign the extrinsic? (Selecting 'No' will prompt you to manually enter the secret key URI for signing, e.g., '//Alice')";
+
+/// Launches the wallet integration for in-browser signing. Blocks until the signature is received.
+///
+/// # Arguments
+/// * `call_data` - The call data to be signed.
+/// * `url` - Chain rpc.
+/// # Returns
+/// * The signed payload, if it exists.
+pub async fn request_signature(call_data: Vec<u8>, url: String) -> anyhow::Result<Option<String>> {
+	let ui = FrontendFromString::new(include_str!("../assets/index.html").to_string());
+
+	let transaction_data = TransactionData::new(url, call_data);
+	// starts server
+	let mut wallet = WalletIntegrationManager::new(ui, transaction_data);
```

Comment:
Maybe using `new_with_address` with an available port or refactoring `new` to be able to pass a port number would be necessary before merging.

Reply:
https://github.com/r0gue-io/pop-cli/pull/371/commits/0e19b8a6cc693a545ffa9cd62eaff9e744127fb6